### PR TITLE
Link panel improvments

### PR DESCRIPTION
**Pull request by brichet**
Improve the link panel layout

Peek 2023-06-21 18-31

### CODE REVIEW

**Comment by trungleduc** [comment]
It looks great @brichet! One small visual thing, can you add a line to split 2 columns of the attributes section?

**Review by martinRenou — APPROVED** [review]
Looks great! There is a minor conflict in the CSS file, otherwise ready to merge!

File: src/linkPanel/widgets/summary.tsx [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+                      icon={deleteIcon}
+                      height={'24px'}
+                      width={'24px'}
+                    />
+                  </Button>
+                </td>
+              </tr>
+            </table>
+          </div>
+        ))} */}
```
Comment (trungleduc) [inline src/linkPanel/widgets/summary.tsx:291]: Do we need to keep it?

File: src/linkPanel/widgets/linking.tsx [hunk 2]
```diff
@@ -0,0 +1,745 @@
+// import { ToolbarRegistry } from '@jupyterlab/apputils';
+// import { ObservableList } from '@jupyterlab/observables';
```
Comment (trungleduc) [inline src/linkPanel/widgets/linking.tsx:2]:
```suggestion
```

File: src/linkPanel/widgets/linking.tsx [hunk 3]
```diff
@@ -0,0 +1,745 @@
...
+import {
+  Button,
+  ReactWidget,
+  UseSignal
+  // Toolbar,
+  // ToolbarButton
```
Comment (trungleduc) [inline src/linkPanel/widgets/linking.tsx:8]:
```suggestion
```

File: src/linkPanel/widgets/linking.tsx [hunk 4]
```diff
@@ -0,0 +1,745 @@
...
+} from '@jupyterlab/ui-components';
+import { Widget } from '@lumino/widgets';
+import * as React from 'react';
+
+import { IGlueSessionSharedModel } from '../../types';
+import { LinkEditorWidget } from '../linkEditorWidget';
+import {
+  ComponentLinkType,
+  // IAdvLinkCategories,
+  // IAdvLinkDescription,
```
Comment (trungleduc) [inline src/linkPanel/widgets/linking.tsx:18]:
```suggestion
```

File: src/linkPanel/widgets/linking.tsx [hunk 5]
```diff
@@ -0,0 +1,745 @@
...
+  ILink,
+  ILinkEditorModel,
+  IDatasets,
+  IDatasetsKeys,
+  // IdentityLinkFunction,
```
Comment (trungleduc) [inline src/linkPanel/widgets/linking.tsx:23]:
```suggestion
```

File: src/linkPanel/widgets/linking.tsx [hunk 6]
```diff
@@ -0,0 +1,745 @@
...
+} from '../types';
+import { ISignal, Signal } from '@lumino/signaling';
+
+export class Linking extends LinkEditorWidget {
+  constructor(options: LinkEditorWidget.IOptions) {
+    super(options);
+
+    this.addClass('glue-LinkEditor-linking');
+
+    // this._selectedAdvLink = { category: '', linkName: '' };
```
Comment (trungleduc) [inline src/linkPanel/widgets/linking.tsx:34]:
```suggestion

```

File: src/linkPanel/widgets/linking.tsx [hunk 7]
```diff
@@ -0,0 +1,745 @@
...
+  //  * Update the advanced link panel when the selected advanced link changed.
+  //  *
+  //  * @param selectedLink - the selected advanced link.
+  //  */
+  // onAdvancedLinkChanged = (
+  //   selectedLink: Private.IAdvancedLinkSelected
+  // ): void => {
+  //   this._selectedAdvLink = selectedLink;
+  //   this.updateAdvancedLink();
+  // };
```
Comment (trungleduc) [inline src/linkPanel/widgets/linking.tsx:102]: This block is for the future advanced link?

Comment (brichet) [reply src/linkPanel/widgets/linking.tsx:102]: Yes, all these blocks are related to the previous advanced link panel and advanced link summary.
I didn't want to drop all of these and start from nothing when we'll implement the advanced links.
But I guess I can remove them from now and I'll watch in commits history if necessary.

Comment (trungleduc) [reply src/linkPanel/widgets/linking.tsx:102]: No worry, let's keep it here for future reference

**Review by trungleduc — APPROVED** [review]

**Comment by trungleduc** [comment]
Thanks @brichet

**Pull request merged**